src/include/miopen/miopen_internal.h

```diff
@@ -0,0 +1,114 @@
+
+#ifdef __clang__
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wextern-c-compat"
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* Begin of Find Mode API */
+
+/*! @brief Version of the Find Mode API for convolutions.
+*
+* The Find Mode API is experimental and therefore subject to change.
+* This macro allows applications to adapt to the future.
+*/
+#define MIOPEN_API_VERSION_CONVOLUTION_FIND_MODE 1
+
+/*! @enum miopenConvolutionFindMode_t
+*
+* * Normal: This is the full Find mode call, which will benchmark all the solvers and return a list.
+*
+* * Fast: Checks the Find-db for an entry. If there is a hit, use that entry. If there is a miss,
+* utilize the Immediate mode fallback. If Start-up times are expected to be faster, but worse GPU
+* performance.
+*
+* * Hybrid: Checks the Find-db for an entry. If there is a hit, use that entry. If there is a miss,
+* use the existing Find machinery. Slower start-up times than Fast Find, but no GPU performance
+* drop.
+*
+* * Fast Hybrid: Checks the Find-db for an entry. If there is a hit, use that entry. If there is a
+* miss, uses the existing Find machinery with skipping slow-compiling kernels. Faster start-up times
+* than Hybrid Find, but GPU performance is a bit worse.
+*
+* * Dynamic Hybrid: This mode is similar to Fast Hybrid, but in case of Find-db miss skips all
+* non-dynamic kernels, thus saving compilation time. Versus Fast Hybrid, we expect similar start-up
+* times but better GPU performance.
+*/
+typedef enum {
+    miopenConvolutionFindModeNormal        = 1,                         /*!< Normal mode */
+    miopenConvolutionFindModeFast          = 2,                         /*!< Fast mode */
+    miopenConvolutionFindModeHybrid        = 3,                         /*!< Hybrid mode */
+    miopenConvolutionFindModeFastHybrid    = 4,                         /*!< Fast Hybrid mode */
+    miopenConvolutionFindModeDynamicHybrid = 5,                         /*!< Dynamic Hybrid mode */
+    miopenConvolutionFindModeDefault = miopenConvolutionFindModeHybrid, /*!< Default setting */
+} miopenConvolutionFindMode_t;
+
+/*! @brief Sets the Find Mode attribute in the convolution descriptor.
+*
+* The subsequent calls of miopenFindConvolutionForwardAlgorithm(),
+* miopenFindConvolutionBakwardDataAlgorithm(), miopenFindConvolutionBakwardDataAlgorithm(),
+* invoked with convDesc, will follow the findMode set by this call.
+*
+* Note that the default Find Mode is set by the MIOPEN_FINE_MODE environment variable,
+* if it is set. If unset, the default is as specified by miopenConvolutionFindModeDefault.
+*
+* @param convDesc   Convolution layer descriptor (input)
+* @param findMode   Find Mode of convDesc (input)
+* @return           miopenStatus_t
+*/
+MIOPEN_EXPORT miopenStatus_t miopenSetConvolutionFindMode(miopenConvolutionDescriptor_t convDesc,
+                                                          miopenConvolutionFindMode_t findMode);
```

Comment:
The API still takes an enum, we need a function to take an `int` instead.

Reply:
Don't care, these functions are not for you.